docs/src/modules/components/AppSearch.js

```diff
@@ -297,6 +297,14 @@ export default function AppSearch() {
                   parseUrl.href = item.url;
                 }
 
+                // TODO: remove this logic once the migration to new structure is done.
+                if (router.asPath.startsWith('/material')) {
+                  parseUrl.href = parseUrl.href.replace(
+                    /(?<!material\/)(getting-started|components|api|customization|guides|discover-more)(\/[^/]+\/)/,
```

Comment:
We can't use the negative lookbehind pattern `?<!`: https://caniuse.com/js-regexp-lookbehind. We got a bunch of people complaining, e.g. https://twitter.com/ludvigerikss0n/status/1473287869242429453, a number more via PM on Twitter.

#30345